src/containers/ConversionForm/index.tsx

```diff
@@ -55,15 +62,46 @@ const ConversionForm = ({
     const [fileToConvert, setFileToConvert] = useState<UploadFile>();
     const [engineSelected, setEngineSelected] = useState<boolean>(false);
     const [isProcessing, setIsProcessing] = useState<boolean>(false);
+    const [fileTypeErrorModalOpen, setFileTypeErrorModalOpen] = useState(false);
 
+    const toggleModal = () => {
+        setFileTypeErrorModalOpen(!fileTypeErrorModalOpen);
+    };
     const toggleProcessing = () => {
         setIsProcessing(!isProcessing);
     };
 
+    const handleEngineChange = (selectedValue: string) => {
+        const selectedEngine = selectedValue as AvailableEngines;
+        setConversionEngine(selectedEngine);
+        setEngineSelected(true);
+    };
+
+    const validateFileType = () => {
+        if (fileToConvert) {
+            const fileExtension =
+                fileToConvert.name.split(".").pop()?.toLowerCase() || "";
```

Comment:
why is the pop uncertain here? split always returns an array

Reply:
ok will remove

Reply:
it's giving `Object is possibly undefined ` without the `?`

Can either leave the `?` or refactor ir

Reply:
is that because `name` might be undefined ? I would probably send in `fileToConvert.name` as an argument in this function ie. 
```const validateFileType = (fileName) => {
              if (fileName) {
                    const fileExtension = fileName.split(".").pop()?.toLowerCase() || "";
```

Reply:
pop returns undefined if the array is empty

Reply:
```
    const validateFileType = (fileName: string) => {
        if (fileName) {
            const fileExtension = fileName.split(".").pop().toLowerCase();
```

gives the same error which is too bad, i would figure the `if (fileName)` where serve as a check.